### ISSUE TITLE
Shared frontend library \w common dependencies

### ISSUE THREAD

**Issue by bzz** [src-d/gitbase-playground#3]
Similar experience to using react-create-app, but instead we will have most of the things FE apps depend on, inside a single library.

It defines:
 - build/lint tasks
 - React
 - Bootstrap with Less

**Comment by smacker** [comment]
Please consider adding redux/redux-thunk/devtools/reselect/router/helmet here too.
Also, it would be nice to have jest working with fetch and localStorage out of the box.

**Comment by carlosms** [comment]
I've been toying with this, and I'd like to share a basic version so we can discuss if this is what we want.

I've made a fork of [create-react-app](https://github.com/facebook/create-react-app) in [carlosms/create-react-app](https://github.com/carlosms/create-react-app). In this fork I've made a few changes to the react-scripts package ([master](https://github.com/carlosms/create-react-app/commits/master)) to add the eslint and prettier conf we use in [code-annotation](https://github.com/src-d/code-annotation).

This forked react-scripts can be used by publishing to npm, or by using a url. So I've made a [GitHub release](https://github.com/carlosms/create-react-app/releases/tag/src-d-react-scripts%400.0.1), and it can be used with this command:

```bash
npx create-react-app --scripts-version https://github.com/carlosms/create-react-app/releases/download/src-d-react-scripts%400.0.1/src-d-react-scripts-0.0.1.tgz testapp
```

Now you can run the new `yarn lint` and `yarn format` commands.

Cons: it might be a bit overkill, and it will require some work to keep up to date with upstream changes.

Before discussing which ones of bootstrap/redux/redux-thunk/devtools/reselect/router/helmet/... should be included, any comments on the basic mechanism? Is this the path we want to follow?

**Comment by dpordomingo** [comment]
I'm not sure if we **really** want to `fork create-react-app`

Instead of forking, does not let `create-react-app` to plug your own stuff?

I never played with it, so I'm just wondering if we could create our own module, add it to our `package.json`, and then just run `yarn build/lint`...

**Comment by carlosms** [comment]
Can you explain your proposal a bit more?

If I understand you correctly, you are proposing to have `node_modules/.bin/apps-scripts` and `node_modules/apps-scripts/eslintrc`? But in that case you still need to add all the dependencies in the main `package.json`, right? (for example for redux, bootstrap, etc).

And I don't know if moving `eslintrc` to an uncommon dir could bring unexpected side effects. For example maybe the editor will not pick it up automatically...

**Comment by smacker** [comment]
yep. Config should be located in the root of a repository. Otherwise, most of text editors won't find it (by default).

**Comment by dpordomingo** [comment]
I wouldn't say that the priority is the editor but the the CI; but since there is a majority I'd vote for the simpler approach too.

**Comment by bzz** [comment]
Let's go for a simpler approach!

**Comment by bzz** [comment]
We can store initial version in https://github.com/src-d/company/tree/master/documentation/apps
